.github/workflows/ui-tests/cypress/integration/deposit_spec.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const defaultUser = Cypress.env('defaultUser')
+const username = defaultUser.username
+const password = defaultUser.password
+const name = defaultUser.name
+const externalAccount = defaultUser.externalAccounts[0]
+
+const depositMsgs = Cypress.env('messages').deposit
+const invalidFeedback = Cypress.env('messages').invalidFeedback
+const formatter = new Intl.NumberFormat('en-US', {
+    maximumFractionDigits: 2,
+  });
+
+const randomInt = (min, max) => {
+    // max is exclusive and the min inclusive
+    return Math.floor(Math.random() * (max-min)) + min
+}
+
+const validPayment = () => {
+    const max = 100
+    const min = 1
+    const num = (Math.random() * max) + min
+    return formatter.format(num)
+}
+
+const validAccountNum = () => {
+    // 10 digit integer
+    const max = 10000000000
+    const min = 1000000000
+    return randomInt(min, max)
+}
+
+const validRoutingNum = () => {
+    // 9 digit integer
+    const max = 1000000000
+    const min = 100000000
+    return randomInt(min, max)
+}
+
+describe('Authenticated default user', function () {
+    beforeEach(function () {
+        cy.loginRequest(username, password)
+        cy.visit('/home')
+    })
+
+    it('sees deposit button', function () {
+        cy.get('#depositSpan').contains('Deposit Funds')
+    })
+
+    it('clicking deposit button makes modal visible', function () {
+        cy.get('#depositFunds').should('not.be.visible')
+        cy.get('#depositSpan').click()
+        cy.get('#depositFunds').should('be.visible')
+    })
+
+    it('sees expected external accounts', function () {
+        cy.get('#depositSpan').click()
+        cy.get('#depositFunds').should('be.visible')
+        cy.get('#accounts').children().first().as('firstOption')
+        cy.get('@firstOption').contains('External Bank')
+        cy.get('@firstOption').contains('9099791699')
+        cy.get('@firstOption').contains('808889588')
```

Comment:
I'm not sure if we need to test exact values here. I'd say it's more important that there's a valid number, instead of asserting exactly what the number is. What do you think?

Reply:
hmm I added it as a sanity check that the test data was correct.  how about changing it to reference the test data specifically, so example: `contains(externalAccount.accountNum)`

Reply:
Yeah that would probably be best! But if that's too much work I don't think it's super important

Reply:
That might be better so we only need to update the values in one place if things change. But I don't think this is too important